### PR TITLE
feat(plugin): connect building search

### PR DESCRIPTION
**Pull request by airslice**
# Overview

This PR connects building search.

## What I've done

- Connect building search with search index & functionality.

## How I tested

Test project [here](https://test.reearth.dev/edit/01gsc8e191z3w20a135dnqjrws). 

Steps:
1. カタログから検索する
2. Select 東京都 -> 西東京市 -> 建築物モデル（西東京市）-> シーンに追加
3. 設定 -> フィルドを追加 -> 3Dタイル -> カメラ
  Warning!!: DO NOT click 保存 / or the data will break the sidebar on other branch -- Kyle-san 
4. 公開 -> カメラ  (fly to the city)
5. 公開 -> Search Data 
6. Do tests.

## Screenshot

image

## Memo

### CODE REVIEW

File: plugin/web/extensions/sidebar/popups/buildingSearch/types.ts [hunk 1]
```diff
@@ -1,22 +1,65 @@
...
+          [key: string]: {
+            count: number;
+            url: string;
+          };
+        };
+      };
+    };
+  };
+  resultsData?: any[];
 };
```
Comment (pyshx) [inline plugin/web/extensions/sidebar/popups/buildingSearch/types.ts:30]: Can you create a separate type for indexes?
It'll be a bit cleaner and we might want to support other type of indexes soon.

Ref:
Screenshot 2023-02-20 at 14 28 21

**Review by pyshx — APPROVED** [review]
LGTM!

File: plugin/web/extensions/sidebar/popups/buildingSearch/components/ResultPanel/ResultItem.tsx [hunk 2]
```diff
@@ -5,38 +5,43 @@ import { Result } from "../../types";
...
-      onSelect([item.id]);
+      onSelect([item]);
     }
   }, [onSelect, item, selected]);
   return (
-    <StyledResultItem onClick={onClick} active={selected.includes(item.id)}>
-      {item.id}
+    <StyledResultItem
+      onClick={onClick}
+      active={!!selected.find(s => s.gml_id === item.gml_id)}
```
Comment (keiya01) [inline plugin/web/extensions/sidebar/popups/buildingSearch/components/ResultPanel/ResultItem.tsx:24]: I guess if `selected` props is huge data, it's better to memorize it.

File: plugin/web/extensions/sidebar/popups/buildingSearch/components/hooks.ts [hunk 3]
```diff
@@ -78,40 +92,188 @@ export default () => {
...
+                  await Promise.all(
+                    cond.values.map(async v => {
+                      await loadDetailData(
+                        `${si.baseURL}/${si.indexRoot.indexes[cond.field].values[v].url}`,
+                      ).then(v => {
+                        condGroup.push(...v.map(item => item.dataRowId));
+                      });
+                    }),
+                  );
+                  condGroups.push(condGroup);
```
Comment (keiya01) [inline plugin/web/extensions/sidebar/popups/buildingSearch/components/hooks.ts:215]: Does it cause N+1 problem?

**Review by keiya01 — APPROVED** [review]

File: plugin/web/extensions/sidebar/popups/buildingSearch/components/ResultPanel/ResultItem.tsx (hunk 2 above)
Comment (airslice) [reply plugin/web/extensions/sidebar/popups/buildingSearch/components/ResultPanel/ResultItem.tsx:24]: Currently `selected` should contain only one item, but might need support multiple even a lot of items in the future. I've updated this part with memorize. Thank you!

File: plugin/web/extensions/sidebar/popups/buildingSearch/components/hooks.ts (hunk 3 above)
Comment (airslice) [reply plugin/web/extensions/sidebar/popups/buildingSearch/components/hooks.ts:215]: I believe this is kind of N + 1 problem, but it's requesting independent csv files from server not a database, so maybe cannot optimize unless there's some union search API provided from server end.

File: plugin/web/extensions/sidebar/popups/buildingSearch/types.ts (hunk 1 above)
Comment (airslice) [reply plugin/web/extensions/sidebar/popups/buildingSearch/types.ts]: Okey sure!

**Review by keiya01 — APPROVED** [review]

**Pull request merged**